### PR TITLE
feat: dataset uses global variables instead of context

### PR DESCRIPTION
**Pull request by rgonalo**

### CODE REVIEW

**Review by pabloge — CHANGES_REQUESTED** [review]

File: toolium/utils/dataset.py [hunk 1]
```diff
@@ -275,14 +282,13 @@ def _infer_param_type(param):
     return new_param
 
 
-def map_param(param, context=None):
+def map_param(param):
```
[the remark this answers is not in the document]
Comment (rgonalo) [reply toolium/utils/dataset.py]: added backwards compatibility

File: toolium/utils/dataset.py [hunk 2]
```diff
@@ -293,33 +299,32 @@ def map_param(param, context=None):
...
-        [TOOLIUM:xxxx] Value from the toolium config in context.toolium_config for the key xxxx (key format is
+        [LANG:xxxx] String from the texts dict in language_config global variable for the key xxxx, using the language
+        specified in language global variable (dot notation is used for keys, e.g. button.label)
+        [POE:xxxx] Definition(s) from the POEditor terms list in poeditor_config global variable for the term xxxx
+        [TOOLIUM:xxxx] Value from the toolium config in toolium_config global variable for the key xxxx (key format is
         section_option, e.g. Driver_type)
-        [CONTEXT:xxxx] Value from the context storage dict for the key xxxx, or value of the context attribute xxxx,
-        if the former does not exist
+        [CONTEXT:xxxx] Value from the behave context storage dict in context global variable for the key xxxx, or value
+        of the behave context attribute xxxx, if the former does not exist
```
[the remark this answers is not in the document]
Comment (rgonalo) [reply toolium/utils/dataset.py]: updated also docs/bdd_integration.rst

File: toolium/utils/dataset.py [hunk 3]
```diff
@@ -275,16 +282,27 @@ def _infer_param_type(param):
     return new_param
 
 
-def map_param(param, context=None):
+def map_param(param, deprecated_context=None):
```
[the remark this answers is not in the document]
Comment (rgonalo) [reply toolium/utils/dataset.py:285]: it is due to deprecated checks, so this warning will be fixed when this code is removed

File: toolium/utils/dataset.py [hunk 4]
```diff
@@ -33,6 +33,13 @@
 base_base64_path = ''
 base_file_path = ''
 
+language = None
+language_terms = None
+project_config = None
+toolium_config = None
+poeditor_terms = None
+behave_context = None
```
Comment (pabloge) [inline toolium/utils/dataset.py:41]: How did you manage to make them all the same length? 😄 😄

**Review by pabloge — APPROVED** [review]

**Pull request merged**